navix/tasks.py

```diff
@@ -53,7 +53,7 @@ def navigation(prev_state: State, action: Array, state: State) -> Array:
     any_reached = jnp.sum(reached)
 
     draw = jax.random.uniform(state.key, ())
```

Comment:
Maybe we can also switch to each individual goal depending on its own marginal distribution?
```suggestion
    draws= jax.random.uniform(state.key, (len(goals.probability),))
```